src/docker/composebuilder/index.go

```diff
@@ -84,10 +78,12 @@ func getServicesDockerCompose(options ApplicationOptions, portReservation *types
 	return result, nil
 }
 
-func getServiceEnvVarEndpoints(options ApplicationOptions, serviceConfigs map[string]types.ServiceConfig, portReservation *types.PortReservation) map[string]*types.ServiceEndpoints {
+func getServiceEnvVarEndpoints(options ApplicationOptions) map[string]*types.ServiceEndpoints {
+	portReservation := types.NewPortReservation()
```

Comment:
This will not reserve ports properly. Each time a service calls `getServiceEnvVarEndpoints`, the starting port number will be reset. The initialization of portReservation needs to be kept in `GetApplicationDockerCompose` and passed to each service that uses `getServiceEnvVarEndpoints`

Reply:
Right below this is where we loop through the services. This function is also only called in one place